libs/datasets/timeseries.py

```diff
@@ -567,18 +583,27 @@ def __post_init__(self):
         assert self.data.index.is_unique
         assert self.data.index.is_monotonic_increasing
         assert self.latest_data.index.names == [CommonFields.LOCATION_ID]
-        if self.provenance is not None:
-            assert isinstance(self.provenance, pd.Series)
-            assert self.provenance.index.names == [CommonFields.LOCATION_ID, PdFields.VARIABLE]
+        assert isinstance(self.provenance, pd.Series)
+        assert self.provenance.index.names == [CommonFields.LOCATION_ID, PdFields.VARIABLE]
 
     def append_regions(
         self, other: "MultiRegionTimeseriesDataset"
     ) -> "MultiRegionTimeseriesDataset":
-        return MultiRegionTimeseriesDataset.from_timeseries_df(
-            pd.concat([self.data, other.data], ignore_index=True)
-        ).append_latest_df(
-            pd.concat(
-                [self.latest_data.reset_index(), other.latest_data.reset_index()], ignore_index=True
+        return (
+            MultiRegionTimeseriesDataset.from_timeseries_df(
+                pd.concat([self.data, other.data], ignore_index=True)
+            )
+            .append_latest_df(
+                pd.concat(
```

Comment:
definitely not necessary, but making these dataframes their own variables would make this block more concise

Reply:
Done in a0bdee18